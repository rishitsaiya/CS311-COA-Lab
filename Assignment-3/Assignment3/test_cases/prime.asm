.data
a:
  11
	.text
main:
	load %x0, $a, %x3
	sub %x4, %x4, %x4
	divi %x3, 2, %x4
	sub %x6, %x6, %x6
	addi %x6, 2, %x6
for:
	bgt %x6, %x4, prime
	div %x3, %x6, %x7
	beq %x0, %x31, notprime
	addi %x6, 1, %x6
	jmp for
prime:
	sub %x10, %x10, %x10
	addi %x10, 1, %x10
	end
notprime:
		sub %x10, %x10, %x10
		subi %x10, 1, %x10
		end
