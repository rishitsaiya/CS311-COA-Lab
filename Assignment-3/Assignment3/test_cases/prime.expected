Hash of the Processor State = -1414219998
